standard/types.md

```diff
@@ -59,10 +59,7 @@ array_type
 
 non_array_type
     : value_type
-    | class_type
-    | interface_type
-    | delegate_type
-    | 'dynamic'
+    | (class_type | interface_type | delegate_type | 'dynamic') nullable_type_annotation?
```

Comment:
Was trying not to end up creating new names like `non_array_non_nullable_reference_type` and `non_array_nullable_reference_type`